diff -rupN libvpsc-orig/isnan.h libvpsc/isnan.h
--- libvpsc-orig/isnan.h	2014-10-16 06:39:05 +0200
+++ libvpsc/isnan.h	2014-10-31 12:51:18 +0100
@@ -42,6 +42,7 @@
 
 #include <cmath>
 #include <cfloat>
+#include <math.h>
 
 #if defined(__isnan)
 # define isNaN(_a) (__isnan(_a))	/* MacOSX/Darwin definition < 10.4 */
@@ -63,7 +64,11 @@
 #elif defined(isfinite)
 # define isFinite(_a) (isfinite(_a))
 #else
-# define isFinite(_a) (std::isfinite(_a))
+	#ifdef COLAANE_MAC
+		# define isFinite(_a) (std::isfinite(_a))
+	#else
+		# define isFinite(_a) (!isNaN((_a) - (_a)))
+	#endif
 #endif
 /* If the above doesn't work, then try (finite(_a) && !isNaN(_a)) or (!isNaN((_a) - (_a))).
  * Also, please report a bug as per http://www.inkscape.org/report_bugs.php,
